servers/src/main/java/tachyon/master/TachyonMaster.java

```diff
@@ -203,168 +188,168 @@ public String getWebBindHost() {
-        throw Throwables.propagate(e);
-      } catch (TTransportException e) {
-        LOG.error(e.getMessage(), e);
-        throw Throwables.propagate(e);
-      }
+  protected void startServingWebServer() {
+    // start web ui
+    mWebServer = new MasterUIWebServer(ServiceType.MASTER_WEB, NetworkAddressUtils.getBindAddress(
+        ServiceType.MASTER_WEB, mTachyonConf), this, mTachyonConf);
+    mWebServer.startWebServer();
+  }
 
-      mMasterMetricsSystem.registerSource(mMasterInfo.getMasterSource());
-      mMasterMetricsSystem.start();
-      mWebServer.addHandler(mMasterMetricsSystem.getServletHandler());
-      mWebServer.startWebServer();
-      LOG.info("Tachyon Master version " + Version.VERSION + " started @ " + mMasterAddress);
-      mMasterServiceServer.serve();
-      LOG.info("Tachyon Master version " + Version.VERSION + " ended @ " + mMasterAddress);
+  protected void startServingRPCServer() {
+    // set up multiplexed thrift processors
+    TMultiplexedProcessor processor = new TMultiplexedProcessor();
+    processor.registerProcessor(mBlockMaster.getProcessorName(), mBlockMaster.getProcessor());
+    processor.registerProcessor(mFileSystemMaster.getProcessorName(),
+        mFileSystemMaster.getProcessor());
+    processor.registerProcessor(mRawTableMaster.getProcessorName(), mRawTableMaster.getProcessor());
+
+    // create master thrift service with the multiplexed processor.
+    mMasterServiceServer = new TThreadPoolServer(new TThreadPoolServer.Args(mTServerSocket)
+        .maxWorkerThreads(mMaxWorkerThreads).minWorkerThreads(mMinWorkerThreads)
+        .processor(processor).transportFactory(new TFramedTransport.Factory())
+        .protocolFactory(new TBinaryProtocol.Factory(true, true)));
+
+    // start thrift rpc server
+    mIsServing = true;
+    mStartTimeMs = System.currentTimeMillis();
+    mMasterServiceServer.serve();
+  }
+
+  protected void stopServing() throws Exception {
+    if (mMasterServiceServer != null) {
+      mMasterServiceServer.stop();
     }
+    if (mWebServer != null) {
+      mWebServer.shutdownWebServer();
+    }
+    mIsServing = false;
   }
 
-  /*
-   * Stop a Tachyon master server.
+  /**
+   * Checks to see if the journal directory is formatted.
+   *
+   * @param journalDirectory The journal directory to check
+   * @return true if the journal directory was formatted previously, false otherwise
+   * @throws IOException
    */
-  public void stop() throws Exception {
-    if (mIsStarted) {
-      mWebServer.shutdownWebServer();
-      mMasterInfo.stop();
-      mJournal.close();
-      mMasterMetricsSystem.stop();
-      mMasterServiceServer.stop();
-      mServerTServerSocket.close();
-      mExecutorService.shutdown();
-      mIsStarted = false;
+  private boolean isJournalFormatted(String journalDirectory) throws IOException {
+    UnderFileSystem ufs = UnderFileSystem.get(journalDirectory, mTachyonConf);
+    if (!ufs.providesStorage()) {
+      return true;
```

Comment:
This is the existing behavior. I added comments and a todo to figure out if this is desirable.